daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -613,4 +604,176 @@ object Transaction {
       fa: FailedAuthorization,
   ) extends TransactionError
 
+  /** Whether `other` is the result of reinterpreting this transaction.
+    *
+    * @note This function is asymmetric.
+    */
+  def isReplayedBy[Nid, Cid, Val](
+      recorded: GenTransaction[Nid, Cid, Val],
+      replayed: GenTransaction[Nid, Cid, Val],
+  )(implicit ECid: Equal[Cid], EVal: Equal[Val]): Either[ReplayMismatch[Nid, Cid, Val], Unit] = {
+    import scalaz.syntax.equal._
+    import scalaz.std.option._
+
+    type Exe = Node.NodeExercises[Nid, Cid, Val]
+
+    def fail(recordedMismatch: Option[Nid], replayMismatch: Option[Nid]) =
+      Left(ReplayMismatch(recorded, recordedMismatch, replayed, replayMismatch))
+
+    @tailrec
+    def loop(
+        nids1: Stream[Nid],
+        nids2: Stream[Nid],
+        stack: List[(Nid, Exe, Stream[Nid], Nid, Exe, Stream[Nid])] = List.empty,
```

Comment:
I'm not a big fan of default arguments. Do we _really_ need it here?
```suggestion
        stack: List[(Nid, Exe, Stream[Nid], Nid, Exe, Stream[Nid])],
```

Reply:
I like using default arguments for accumulator of recursive function, specially for this kind of functions that are in the scope of another function and are called only once. This makes clear what is the base case of the recursion, without having to look at the call.